docs/reference/api/widgets/table.rst

```diff
@@ -1,55 +1,112 @@
+            ("Tricia McMillan", 38),
+        ]
+    )
+
+    # Get the details of the first item in the data:
+    print(f"{table.data[0].name} is age {table.data[0].age}")
+
+    # Append new data to the table
+    table.data.append(("Zaphod Beeblebrox", 47))
+
+You can also specify data for a Table using a list of dictionaries. This allows you to
+store data in the data source that won't be displayed in the table. It also allows you
+to control the display order of columns independent of the storage of that data.
+
 .. code-block:: python
 
     import toga
 
-    table = toga.Table(['Heading 1', 'Heading 2'])
+    table = toga.Table(
+        headings=["Name", "Age"],
+        data=[
+            {"name": "Arthur Dent", "age": 42, "planet": "Earth"},
+            {"name", "Ford Prefect", "age": 37, "planet": "Betelgeuse Five"},
+            {"name": "Tricia McMillan", "age": 38, "plaent": "Earth"},
+        ]
+    )
+
+    # Get the details of the first item in the data:
+    print(f"{table.data[0].name}, who is age {table.data[0].age}, "
+          f"is from {table.data[0].planet}")
+
+The attribute names used on each row of data (called "accessors") are created
+automatically from the headings that you provide. If you want to use different
+attributes, you can override them by providing an ``accessors`` argument. In this
+example, the table will use "Name" as the visible header, but internally, the attribute
+"character" will be used:
 
-    # Append to end of table
-    table.data.append('Value 1', 'Value 2')
+.. code-block:: python
 
-    # Insert to row 2
-    table.data.insert(2, 'Value 1', 'Value 2')
+    import toga
 
-        Examples:
-            >>> headings = ['Head 1', 'Head 2', 'Head 3']
-            >>> data = []
-            >>> table = Table(headings, data=data)
+    table = toga.Table(
+        headings=["Name", "Age"],
+        accessors={"Name", 'character'},
+        data=[
+            {"character": "Arthur Dent", "age": 42, "planet": "Earth"},
+            {"character", "Ford Prefect", "age": 37, "planet": "Betelgeuse Five"},
+            {"name": "Tricia McMillan", "age": 38, "plaent": "Earth"},
+        ]
+    )
 
-            Data can be in several forms. A list of dictionaries, where the keys match
-            the heading names:
+    # Get the details of the first item in the data:
+    print(f"{table.data[0].character}, who is age {table.data[0].age}, "
+          f"is from {table.data[0].planet}")
 
-            >>> data = [{'head_1': 'value 1', 'head_2': 'value 2', 'head_3': 'value3'}),
-            >>>         {'head_1': 'value 1', 'head_2': 'value 2', 'head_3': 'value3'}]
+The value provided by an accessor is interpreted as follows:
 
-            A list of lists. These will be mapped to the headings in order:
+* If the value is a :any:`Widget`, that widget will be displayed in the cell. Note that
+  this is currently a beta API, is currently only supported on macOS, and may change in
+  future.
```

Comment:
This was previously in a notes section; I agree it's worth flagging as a beta feature here, I think having the separate notes block that explains exactly what "beta" means in this context is worthwhile.
```suggestion
* If the value is a :any:`Widget`, that widget will be displayed in the cell. Note that
  this is currently a beta API; see the :ref:`Notes section <table-notes>` for details.
```

(with a notes section describing macOS limitations, etc)

Reply:
I know we normally put OS-specific notes in a separate section, but this is a feature which quite possibly will never be implemented on any other platform, so I think that should be stated up front.

Reply:
I agree that it should be stated up front that this is beta, and will almost certainly undergo a radical change before becoming final. What I'm suggesting is that we make that a general comment about API stability in the "main" docs, and leave the platform-specificity to the notes section. However, even if we do mention the macOS specificity here, we've got a general pattern of "platform-specific weirdness is in the notes section"; so I think we should have a Notes section highlighting the macOS restriction.

Reply:
OK, done.